### PR TITLE
[blog] Introducing Vue 3 & Web Components in Hue Query Editor

### PR DESCRIPTION
**Pull request by sreenaths**
## What changes were proposed in this pull request?
Blogpost on Introducing Vue 3 & Web Components in Hue Query Editor

## How was this patch tested?
Using `hugo serve`

### CODE REVIEW

File: docs/gethue/content/en/posts/2021-03-04-vue3-migration.md [hunk 1]
```diff
@@ -0,0 +1,269 @@
+---
+title: Introducing Vue 3 & Web Components in Hue Query Editor
+author: Hue Team
+type: post
+date: 2021-03-04T00:00:00+00:00
+url: /blog/vue3-migration
```
Comment (romainr) [inline docs/gethue/content/en/posts/2021-03-04-vue3-migration.md:6]: URL is not SEO or humane explicit

File: docs/gethue/content/en/posts/2021-03-04-vue3-migration.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+sf_right_sidebar:
+  - Sidebar-1
+sf_caption_position:
+  - caption-right
+sf_remove_promo_bar:
+  - 1
+ampforwp-amp-on-off:
+  - default
+categories:
+  - Version 4.9
```
Comment (romainr) [inline docs/gethue/content/en/posts/2021-03-04-vue3-migration.md:42]: This is 4.10

File: docs/gethue/content/en/posts/2021-03-04-vue3-migration.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+
+### Using official Vue 3 web component wrapper
+
+The official wrapper must be very similar to our implementation. That said, the exact function signature of the official wrapper is unknown at this point. Keep a check on [this page](https://www.npmjs.com/package/vue3-webcomponent-wrapper) for future updates.
+
+## And that's it!
+
+Hue moved from using pretty old technologies to the forefront of Web interface development with Vue 3 and the component wrapper. We strongly believe component programming is a very effective development paradigm with its isolation and easy sharing. In the next episode we will demo how it all integrates with the new Hue 5 API.
+
+~ Sreenath & Romain from the Hue Team
```
Comment (romainr) [inline docs/gethue/content/en/posts/2021-03-04-vue3-migration.md:269]: No need to put my name, remove it ;)

File: docs/gethue/content/en/posts/2021-03-04-vue3-migration.md [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+_Note: One main blocker preventing the official wrapper for upgrading was the lack of shadow-root CSS injection in Vue 3 build tooling. As we could live without shadow in Hue this was not an issue and port took hardly a day._
+
+Using our component wrapper is pretty easy. It can be installed using `npm i --save vue3-webcomponent-wrapper`. Once installed following snippet show how to create a custom tag named `my-component`.
+
+    import { createApp, h } from "vue";
+    import wrapper from "vue3-webcomponent-wrapper";
+    import MyComponent from "./components/MyComponent.vue";
+
+    const webComponent = wrapper(MyComponent, createApp, h);
+    window.customElements.define("my-component", webComponent);
```
Comment (romainr) [inline docs/gethue/content/en/posts/2021-03-04-vue3-migration.md:257]: Same comment as in other PR about showing the before/with/after life cycle

File: docs/gethue/content/en/posts/2021-03-04-vue3-migration.md [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+
+    defineComponent(() => {
+      return {
+        data: 1
+      };
+    });
+
+### Our component template
+
+At the end we decided to go with object-style syntax, and a combination of Options & Composition API where setup hook acts as the setup function. **We decided to mix and match as defining static items like props or components is easy with Options API and dynamic items like reactive provide & inject is easy with Composition API. Moreover, many of the Vue 3 documentation uses this style.** After going through various sources we created the following template for a component. It provides various options (i.e, components, directives, props etc ) that can be used to define a component, the order they could be used, and how each part interacts. This acts as a base for all our components.
```
Comment (romainr) [inline docs/gethue/content/en/posts/2021-03-04-vue3-migration.md:145]: Nice!

**Review by romainr — APPROVED** [review]

File: docs/gethue/content/en/posts/2021-03-04-vue3-migration.md (hunk 2 above)
Comment (sreenaths) [reply docs/gethue/content/en/posts/2021-03-04-vue3-migration.md:42]: Done

File: docs/gethue/content/en/posts/2021-03-04-vue3-migration.md (hunk 3 above)
Comment (sreenaths) [reply docs/gethue/content/en/posts/2021-03-04-vue3-migration.md]: Without you the blog wouldn't be good enough 🙂

**Comment by romainr** [comment]
Big nice one!

It is live now https://gethue.com//posts/

**Pull request merged**